### DISCUSSION TITLE
StatusLive公共反代使用说明

### DISCUSSION THREAD

**Discussion by freejishu** [freejishu/StatusLive#15] [Announcements]
本文将帮助你申请一个UptimeRobot API的公共反代，并将其直接用于StatusLive，实现真正的开箱即用。

本服务由开发者freejishu提供，欢迎赞助ouo。

## Public Token 生成须知

1、本服务由 freejishu的美丽世界(www.freejishu.com) 提供，属于公益性服务。我们会努力保障SLA，但是不做出承诺。

2、本服务过程中将会使用您提交的UptimeRobot API Key访问UptimeRobot，并将结果进行处理后返回给访问者。 根据访问需要，我们会使用程序处理其中的数据（抹去部分敏感字段信息），但不会将数据进行记录亦或者进行分析（用于缓解服务器访问压力的缓存唯一除外）。

3、反代系统已配置缓存机制，缓存时间为10秒。公平起见，每个Public Token访问接口的频率上限10000次/天（后期亦有可能根据实际情况调整数值），若您预期将超过此请求量，请自行搭建服务。 我们有权对超过上限的请求进行拒绝、屏蔽或封禁。

4、尽管数据在传输过程中严格进行加密，但是考虑到互联网的不确定性，建议使用Read-Only API Key（只读Key）生成Public Token。

5、生成的Public Token只会在本页面显示一次，请妥善保存。若Public Token丢失，请使用生成时留下的邮箱联系我们（Mail：i#freejishu.com，请将#换为@）。

6、对本规则的未尽事宜，我们保留最终解释权。（也没啥好解释的，公益服务大家好好珍惜）

### 2、输入信息。这里的API Key建议使用Read-Only API Key（只读Key）。
![image](https://user-images.githubusercontent.com/11630435/187026873-0b545095-2d94-4b7d-94e4-437f76970279.png)

### 3、点击立即创建即可。
![image](https://user-images.githubusercontent.com/11630435/187026948-4d60a875-6519-49ae-a4ab-64c0f298b96d.png)

### 4、将 API 链接复制到 conf.json 的 config_proxy_link 字段中。
    ```
    {
        ...

        "config_mode": 2,  //使用公共反代则此选项保持2
        "config_readonly_apikey": "USE_SERVER_APIKEY", //保持USE_SERVER_APIKEY不变
        "config_proxy_link": "https://statuslive.freejishu.com/core/?publickey=38401bea-1e90-xxxx-xxxx-b7xxxxxxxx47", //隐私模式用，反代访问路径

         ...
    }
    ```

完成！Enjoy it！

**Comment by hifocus** [comment]
Hi, 一些不成熟的意见
简单看了一下，用 PHP 处理的目的就是去除敏感信息；但是同时作为 UptimeMonitor，提供的 API 非常可以理解地没办法保障 SLA；
那么有没有可能用 CloudFlare Workers 写，简单处理也可以去除敏感信息；每个有 CloudFlare 账号的人都可以有每天 1M 请求的 quota，部署也非常方便，SLA 之类的完全自己负责，同时也没有维护/服务器负载的压力；有没有可能通过那样实现呢？

**Reply by hifocus** [reply]
btw accordingly vercel.app 在国内被屏蔽了（

**Reply by freejishu** [reply]
意见很好：）

之前考虑了Vercel等这类平台，但是有个缺点是Serverless平台普遍不支持“缓存”的机制，无法存留数据；但是Cloudflare的Workers是个例外。2.0.2发布之前我会再行研究。

os：暂时用下vercel，回头会在国内部署，但是如果Workers体验更好也许我自己也会转向Workers（

**Comment by rwv** [comment]
首先非常感谢这个项目♥！原本准备自己写一个的，发现已经有这么棒的项目了实在很棒！
可以使用 Cloudflare 的 Pages Function，同时解决静态文件和 API 反代的问题。
https://developers.cloudflare.com/pages/how-to/refactor-a-worker-to-pages-functions/

**Reply by freejishu** [reply]
准确的说，已经基于Cloudflare Worker写了一版反代的代码，但是CROS一直没有调好，所以就搁置了一下 ;w;
若有兴趣，也可以试着写一个然后贡献一下^^

**Comment by Masterain98** [comment]
反代服务似乎掉线了？

**Reply by freejishu** [reply]
确实掉线了，前两天重启了下服务器，这个服务没有自启，现已恢复
